apache/zeppelin
# ZEPPELIN-1935 Add jceks stored password support for jdbc interpreter

### What is this PR for?
Add support for jceks stored password instead of using password in clear text. This security enhancement prevents any user from reading clear passwords from interpreter json using shell/py/spark etc. #1315 is the parent PR which fixed similar a issue in Active Directory (shiro.ini config).

### What type of PR is it?
Improvement

### Todos

### What is the Jira issue?
ZEPPELIN-1935

### How should this be tested?
Create a keystore file using the hadoop credential commandline, for this the hadoop commons should be in the classpath.

`hadoop credential create jdbc.password -provider jceks://file/user/zeppelin/conf/zeppelin.jceks`

Use the jceks file and the key to configure jdbc interpreter. Example interpreter setting:

#### Following are the supported settings for backward compatibility

- Settings with username and jceks
```
default.driver	org.postgresql.Driver
default.jceks.credentialKey	jdbc.password
default.jceks.file	jceks://file/tmp/zeppelin.jceks
default.url	jdbc:postgresql://rkamath-local-1:5432/
default.user	rk-user
```

- Settings with user name and clear text password
```
default.driver	org.postgresql.Driver
default.url	jdbc:postgresql://rkamath-local-1:5432/
default.user	rk-user
default.password	password1
```

### Screenshots (if appropriate)
<img width="1392" alt="screen shot 2017-01-10 at 7 02 12 pm" src="https://cloud.githubusercontent.com/assets/2031306/21808016/5e602982-d767-11e6-88f9-3d15b9a7f0b8.png">



### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? yes
